x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -335,15 +336,17 @@ export class TaskRunner {
           };
         }
       ),
-      runAt: resolveErr<Date, Error>(runAt, () =>
-        getNextRunAt(
-          new Date(),
-          // if we fail at this point we wish to recover but don't have access to the Alert's
-          // attributes, so we'll use a default interval to prevent the underlying task from
-          // falling into a failed state
-          FALLBACK_RETRY_INTERVAL
-        )
-      ),
+      runAt: resolveErr<Date | undefined, Error>(runAt, err => {
+        return isAlertSavedObjectNotFoundError(err, alertId)
+          ? undefined
```

Comment:
Tested this locally and can confirm the task gets removed (stops running each 5 minutes). Though there is still a log like below that is made. This comes from a line above (~332), it could follow task manager's approach and become a debug message in this situation.

```
Executing Alert "0086e8a2-f588-4cd9-82d4-5978ee33c55a" has resulted in Error: Saved object [alert/0086e8a2-f588-4cd9-82d4-5978ee33c55a] not found
```

Reply:
Was there a separate issue for this - that the task wasn't actually removed in this case and so kept running forever?  I didn't see one in a quick pass of the issues, maybe we never created an issue for it.

Reply:
The issue is fairly new, I figured we'd re-use the original issue to fix the removal of the task as well (https://github.com/elastic/kibana/issues/42477#issuecomment-610460266).